src/index/Index.h

```diff
@@ -506,12 +524,16 @@ class Index {
 
   void passContextFileIntoVector(const string& contextFile, TextVec& vec);
 
-  // no need for explicit instatiation since this function is private
   template <class MetaDataDispatcher>
-  std::optional<typename MetaDataDispatcher::WriteType> createPermutationImpl(
-      const string& fileName, const TripleVec& vec, size_t c0, size_t c1,
-      size_t c2);
-  template <class MetaDataDispatcher, class Comparator1, class Comparator2>
+  std::optional<std::pair<typename MetaDataDispatcher::WriteType,
+                          typename MetaDataDispatcher::WriteType>>
+  createPermutationPairImpl(const string& fileName1, const string& fileName2,
+                            const Index::TripleVec& vec, size_t c0, size_t c1,
+                            size_t c2);
+
+  pair<FullRelationMetaData, BlockBasedRelationMetaData> writeSwitchedRel(
+      ad_utility::File& out, off_t lastOffset, Id currentRel,
+      ad_utility::BufferedVector<array<Id, 2>>* buffer);
```

Comment:
I copied the File& out parameter from the original method, and added the Argument *buffer because I think the same about non-const references. I will update it to be at least consistent on a per-function level.